vkcs/cdn/resource_resource.go

```diff
@@ -0,0 +1,445 @@
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/cdn/resource_resource"
+	resource_validators "github.com/vk-cs/terraform-provider-vkcs/vkcs/cdn/resource_resource/validators"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/clients"
+	fwutils "github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/framework/utils"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/cdn/v1/resources"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/util/errutil"
+)
+
+var (
+	_ resource.Resource                     = (*resourceResource)(nil)
+	_ resource.ResourceWithConfigure        = (*resourceResource)(nil)
+	_ resource.ResourceWithConfigValidators = (*resourceResource)(nil)
+	_ resource.ResourceWithImportState      = (*resourceResource)(nil)
+	_ resource.ResourceWithValidateConfig   = (*resourceResource)(nil)
+)
+
+func NewResourceResource() resource.Resource {
+	return &resourceResource{}
+}
+
+type resourceResource struct {
+	config clients.Config
+}
+
+func (r *resourceResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cdn_resource"
+}
+
+func (r *resourceResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_resource.ResourceResourceSchema(ctx)
+}
+
+func (r *resourceResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *resourceResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	return []resource.ConfigValidator{
+		resource_validators.ConflictingEnabled(
+			path.MatchRoot("options").AtName("fetch_compressed"),
+			path.MatchRoot("options").AtName("gzip_on"),
+		),
+		resource_validators.ConflictingEnabled(
+			path.MatchRoot("options").AtName("forward_host_header"),
+			path.MatchRoot("options").AtName("host_header"),
+		),
+		resource_validators.ConflictingEnabled(
+			path.MatchRoot("options").AtName("ignore_query_string"),
+			path.MatchRoot("options").AtName("query_params_blacklist"),
+			path.MatchRoot("options").AtName("query_params_whitelist"),
+		),
+	}
+}
+
+func (r *resourceResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var config resource_resource.ResourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &config)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	sslCertType := config.SslCertificate.SslCertificateType.ValueString()
+	if sslCertType == string(resource_resource.SslCertificateProviderTypeOwn) && config.SslCertificate.Id.IsNull() {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("ssl_certificate").AtName("id"),
+			"Missing Attribute Value",
+			"value must be configured when `ssl_certificate.type` is \"own\"",
```

Comment:
grave accent?

Reply:
I think we usually use backtick to specify other fields or resources because it renders in the specific way, and it may be conventional , but I'm not sure.